lighthouse-core/audits/viewport.js

```diff
@@ -38,8 +39,15 @@ class Viewport extends Audit {
    * @return {!AuditResult}
    */
   static audit(artifacts) {
-    const hasMobileViewport = typeof artifacts.Viewport === 'string' &&
-        artifacts.Viewport.includes('width=');
+    if (typeof artifacts.Viewport !== 'string') {
+      return Viewport.generateAuditResult({
+        rawValue: false
+      });
+    }
+
+    const viewportProps = Parser.parseMetaViewPortContent(artifacts.Viewport).validProperties;
```

Comment:
now that we have a parser, it might be nice to also warn if there are `unknownProperties` or `invalidValues` values. If that's the case, add a debugString to the artifact.

```
let debugString = '';
const parsedProps = Parser.parseMetaViewPortContent(artifacts.Viewport);

if (Object.keys(parsedProps.unknownProperties).length) {
  debugString += `Invalid properties found: ${parsedProps.unknownProperties}`;
}
if (Object.keys(parsedProps.invalidValues).length) {
  debugString += `Invalid values found: ${parsedProps.invalidValues}`;
}

const viewportProps = parsedProps.validProperties;
const hasMobileViewport = viewportProps.width || viewportProps['initial-scale'];

Viewport.generateAuditResult({
  rawValue: !!hasMobileViewport
  debugString
});
```